From: Gleb Pesin <dormancygrace@gmail.com>
Subject: [PATCH] flatbuffers: build the host compiler as C++17

The host GCC rejects the C++0x mode of flatbuffers 1.x with the
abseil-style string_view it now selects; C++17 builds it.
---
diff --git a/flatbuffers/CMakeLists.txt b/flatbuffers/CMakeLists.txt
index dbb2325c..cbbc0f1d 100644
--- a/flatbuffers/CMakeLists.txt
+++ b/flatbuffers/CMakeLists.txt
@@ -216,7 +216,7 @@ elseif(CMAKE_COMPILER_IS_GNUCXX)
       "${CMAKE_CXX_FLAGS} -std=gnu++11")
   else(CYGWIN)
     set(CMAKE_CXX_FLAGS
-      "${CMAKE_CXX_FLAGS} -std=c++0x")
+      "${CMAKE_CXX_FLAGS} -std=c++17")
   endif(CYGWIN)
   set(CMAKE_CXX_FLAGS
     "${CMAKE_CXX_FLAGS} -Wall -pedantic  -Wextra ")
@@ -237,7 +237,7 @@ elseif(CMAKE_COMPILER_IS_GNUCXX)
 
 elseif(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
   set(CMAKE_CXX_FLAGS
-      "${CMAKE_CXX_FLAGS} -std=c++0x -Wall -pedantic  -Wextra -Wno-unused-parameter")
+      "${CMAKE_CXX_FLAGS} -std=c++17 -Wall -pedantic  -Wextra -Wno-unused-parameter")
   set(FLATBUFFERS_PRIVATE_CXX_FLAGS "-Wold-style-cast")
   if(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.8)
     list(APPEND FLATBUFFERS_PRIVATE_CXX_FLAGS "-Wimplicit-fallthrough" "-Wextra-semi" "") # enable warning
